### PR TITLE
TEAM CRUD Operations

### PR DESCRIPTION
**Pull request by TadisOn**
Created Team Model, Services, Repository, Controller. Implemented and tested all CRUD operations. Also added "AddPlayer" methods which adds players, but it doesn't work for now

### CODE REVIEW

File: src/api/TourneyRent/TourneyRent.BusinessLogic/Services/TeamService.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        public async Task UpdateTeam(Team team)
+        {
+            await _teamRepository.UpdateTeam(team);
+        }
+
+        public async Task DeleteTeam(Team team)
+        {
+            await _teamRepository.DeleteTeam(team);
+        }
+
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.BusinessLogic/Services/TeamService.cs:45]: Most of these methods should do more. Refer to comments below

File: src/api/TourneyRent/TourneyRent.DataLayer/TourneyRent.DataLayer.csproj [hunk 2]
```diff
@@ -6,10 +6,6 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
-  <ItemGroup>
-    <Folder Include="Migrations\" />
-  </ItemGroup>
-
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.DataLayer/TourneyRent.DataLayer.csproj:12]: Does this remove the folder? Revert this change if possible

File: src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            }
+
+            var teamRead = _mapper.Map<TeamRead>(team);
+            return Ok(teamRead);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TeamRead>>> GetAllTeams()
+        {
+            var teams = await _teamService.GetAllTeams();
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs:43]: Async methods should have a **Async** suffix

File: src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            {
+                if(await _teamService.GetTeamById(id) == null)
+                {
+                    return NotFound();
+                }
+
+                throw;
+            }
+
+            return NoContent();
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs:77]: This could be a service method. Furthermore this exception should not be handled here.

File: src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            var team = await _teamService.GetTeamById(teamId);
+            if(team == null)
+            {
+                return NotFound();
+            }
+
+            var player = team.Players.FirstOrDefault(p => Convert.ToInt32(p.Id) == id);
+            if(player==null)
+            {
+                return NotFound(); ;
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs:113]: Double ;;

File: src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+            }
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs:114]: Most of this logic could be in the service. Also I think it would be better to throw an Exception (in the service) that inherits from BaseDomainException and then you could do something like this in the controller:
````
try {
 // Do stuff
} catch (NotFoundException) {
 return NotFound();
}
````

Or if possible you could catch that in the GlobalExceptionMiddleware and that way we would not need to return NotFound() anywhere else in the code. Although I am not sure if that middleware catches non async methods.

File: src/api/TourneyRent/TourneyRent.Presentation.Api/TourneyRent.Presentation.Api.csproj [hunk 7]
```diff
@@ -9,6 +9,7 @@
   <ItemGroup>
     <PackageReference Include="AutoMapper" Version="12.0.1" />
     <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="12.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.JsonPatch" Version="7.0.3" />
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.Presentation.Api/TourneyRent.Presentation.Api.csproj:12]: Where is this used?

File: src/api/TourneyRent/TourneyRent.Presentation.Api/Views/Team/TeamRead.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
+﻿using TourneyRent.DataLayer.Models;
+
+namespace TourneyRent.Presentation.Api.Views.Team
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.Presentation.Api/Views/Team/TeamRead.cs:3]: Namespace (and folder) should be plurar

File: src/api/TourneyRent/TourneyRent.Presentation.Api/Views/Team/TeamRead.cs [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+{
+    public class TeamRead
```
Comment (jusrus01) [inline src/api/TourneyRent/TourneyRent.Presentation.Api/Views/Team/TeamRead.cs:5]: I think it would be better to add **View** suffix for these e.g. TeamView, just to make it more clear that this class should be used in the Presentation layer.

**Review by jusrus01 — CHANGES_REQUESTED** [review]

File: src/api/TourneyRent/TourneyRent.DataLayer/TourneyRent.DataLayer.csproj (hunk 2 above)
Comment (TadisOn) [reply src/api/TourneyRent/TourneyRent.DataLayer/TourneyRent.DataLayer.csproj:12]: image
I don't see it in my code

Comment (jusrus01) [reply src/api/TourneyRent/TourneyRent.DataLayer/TourneyRent.DataLayer.csproj:12]: Red color means that it was removed by you, so it is expected that you can't see it. I guess it's fine, I'll just add it later again if that folder disappears.

File: src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs (hunk 3 above)
Comment (TadisOn) [reply src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs:43]: As I'm seeing, most of these methods have  waiting for some response or am I understanding it wrong?

Comment (jusrus01) [reply src/api/TourneyRent/TourneyRent.Presentation.Api/Controllers/TeamController.cs:43]: I just meant that methods should be named like this: _teamService.GetAllTeamsAsync(), just like you used _context.SaveChangesAsync(). However, controller methods don't have to have that naming convention, since there is not going to be any consumer in the codebase.

**Comment by jusrus01** [comment]
Also request a review through Github, so I would know that there won't be anymore changes coming

**Comment by jusrus01** [comment]
We can merge after you do those things with exceptions. Front-end will show notification messages when you do it.

**Comment by matas01** [comment]
puikiai

**Review by jusrus01 — APPROVED** [review]
